### PR TITLE
[WIP] update keepdim params for 1D scale factor

### PR DESCRIPTION
**Pull request by namgyu-youn**
**Overview**:
In `_choose_scale_float8`, the per-tensor quantization case (`len(block_size) == 0`) uses `tensor.amax(keepdim=True)` while `_choose_qparams_affine` uses `torch.amax(..., keepdim=False)` for the same purpose.

This PR aligns `_choose_scale_float8` with `_choose_qparams_affine` by using `tensor.amax(keepdim=False)` for 1D scale factor.

**Related Issue/PR**: #3324

**Test Plan**: test/quantization/test_quant_primitives.py

### CODE REVIEW

**Comment by namgyu-youn** [comment]
Local test code:
```python
import torch
from torchao.quantization.quant_primitives import _choose_scale_float8, _choose_scale_float8_old

a = torch.randn(4, 4)
scale = _choose_scale_float8(a, block_size=(4, 1))  # keepdim = False
print(scale)

scale_old = _choose_scale_float8_old(a, block_size=(4, 1))  # keepdim = True
print(scale_old)
```

And the result is (same):
```
tensor([[0.0023, 0.0044, 0.0040, 0.0027]])
tensor([[0.0023, 0.0044, 0.0040, 0.0027]])
```

**Comment by jerryzh168** [comment]
this is not enough I think there are https://github.com/pytorch/ao/blob/2ff1eb2e356275cfbe46832327387d382c72945d/torchao/quantization/quant_primitives.py#L2320-L2323 that will expand the dimension of scale

I actually tried this locally, and doesn't seem to work very well. I think we can leave this for now, but one thing that is useful is to remove the need to do https://github.com/pytorch/ao/blob/2ff1eb2e356275cfbe46832327387d382c72945d/torchao/quantization/quant_primitives.py#L2332, I don't know why we need this if float8 is already using a scale that's matching the dimension of input: https://github.com/pytorch/ao/blob/2ff1eb2e356275cfbe46832327387d382c72945d/torchao/quantization/quant_primitives.py#L2318-L2323

maybe try removing calls to `_maybe_expand_scale_to_tensor_shape` in code and make sure all tests still pass is a good task to work on 


also another thing we can do is to simplify the implementation of https://github.com/pytorch/ao/blob/2ff1eb2e356275cfbe46832327387d382c72945d/torchao/float8/inference.py#L130 since for float8 scale is always matching the dim of input
